### PR TITLE
bugfix for nested maps with vecbuilder issue

### PR DESCRIPTION
**Pull request by jjthomas**
should resolve @rahulpalamuttam issues with birth analysis

### CODE REVIEW

File: weld_rt/cpp/vb.cpp [hunk 1]
```diff
@@ -74,7 +74,11 @@ extern "C" void weld_rt_new_vb_piece(void *v, work_t *w) {
     cur_piece->size = 0;
     cur_piece->capacity = vb->starting_cap;
   } else {
-    cur_piece->data = (uint8_t *)vb->fixed_vector + vb->elem_size * w->cur_idx;
+    // If this is the initial piece (i.e. the one created right after the vb is created),
+    // we want to use an offset of 0. Only for pieces created for new tasks should an
+    // offset of w->cur_idx be used.
+    cur_piece->data = (uint8_t *)vb->fixed_vector +
+      (is_init_piece ? 0 : vb->elem_size * w->cur_idx);
```
Comment (rahulpalamuttam) [inline weld_rt/cpp/vb.cpp]: More of a style thing here - but are we fine with using ternaries vs an explicit if else?

Comment (jjthomas) [reply weld_rt/cpp/vb.cpp]: yeah i think in this case it's pretty straightforward and shortens the code, i agree in cases where you have nested ternaries and stuff it's pretty annoying

**Comment by rahulpalamuttam** [comment]
Looks good.
We can also probably include the nested map test case. 
|v1: vec[vec[i8]]| map(v1, |v2:vec[i8]| map(v2, |e:i8| e))

**Comment by sppalkia** [comment]
Thank you James!

**Pull request merged**